lib/board.rb

```diff
@@ -77,21 +79,46 @@ def place(ship, coord_array)
+  def render(show_ship = false)
+    unless show_ship == true
+      return row_1_render + rest_of_rows.join + "\n"
+    else
+      return row_1_render + rest_of_rows_show_ship_true.join + "\n"
+    end
   end
 
-  def render_true_helper
-       "  1 2 3 4 \nA #{@cells["A1"].render(true)} #{@cells["A2"].render(true)} #{@cells["A3"].render(true)} #{@cells["A4"].render(true)} \nB #{@cells["B1"].render(true)} #{@cells["B2"].render(true)} #{@cells["B3"].render(true)} #{@cells["B4"].render(true)} \nC #{@cells["C1"].render(true)} #{@cells["C2"].render(true)} #{@cells["C3"].render(true)} #{@cells["C4"].render(true)} \nD #{@cells["D1"].render(true)} #{@cells["D2"].render(true)} #{@cells["D3"].render(true)} #{@cells["D4"].render(true)} \n"
+  # def render_helper
+  #   row_1_render + rest_of_rows.join + "\n"
+  #    # "  1 2 3 4 \nA #{@cells["A1"].render} #{@cells["A2"].render} #{@cells["A3"].render} #{@cells["A4"].render} \nB #{@cells["B1"].render} #{@cells["B2"].render} #{@cells["B3"].render} #{@cells["B4"].render} \nC #{@cells["C1"].render} #{@cells["C2"].render} #{@cells["C3"].render} #{@cells["C4"].render} \nD #{@cells["D1"].render} #{@cells["D2"].render} #{@cells["D3"].render} #{@cells["D4"].render} \n"
+  # end
+  #
+  # def render_true_helper
+  #   row_1_render + rest_of_rows_show_ship_true.join + "\n"
+  #      # "  1 2 3 4 \nA #{@cells["A1"].render(true)} #{@cells["A2"].render(true)} #{@cells["A3"].render(true)} #{@cells["A4"].render(true)} \nB #{@cells["B1"].render(true)} #{@cells["B2"].render(true)} #{@cells["B3"].render(true)} #{@cells["B4"].render(true)} \nC #{@cells["C1"].render(true)} #{@cells["C2"].render(true)} #{@cells["C3"].render(true)} #{@cells["C4"].render(true)} \nD #{@cells["D1"].render(true)} #{@cells["D2"].render(true)} #{@cells["D3"].render(true)} #{@cells["D4"].render(true)} \n"
+  # end
+
+  def row_1_render
+    "  " + (1..@num_count).to_a.map { |num| num.to_s.concat(" ") }.join
   end
 
-  def render(show_ship = false)
-    unless show_ship == true
-      return render_helper
-    else
-      return render_true_helper
+  def board_groups
+    (1..@letter_count).to_a.map do |i|
+      @cells.keys[((@num_count*i)-@num_count)..((@num_count*i)-1)]
+    end
+  end
+
+  def rest_of_rows
+    board_groups.map do |group|
+      "\n" + split_letter(group)[0] + " " +
+      @cells.values_at(*group).map { |cell_object| cell_object.render + " " }.join
     end
   end
 
+  def rest_of_rows_show_ship_true
+    board_groups.map do |group|
+      "\n" + split_letter(group)[0] + " " +
+      @cells.values_at(*group).map { |cell_object| cell_object.render(true) + " " }.join
```

Comment:
just took a quick look at what the * means and WOW, that's awesome. I'd love to walk through this logic with you to make sure I understand it all. This is great and will help once we hit IT4

Reply:
I think we can use this logic to refactor build cells, let's take a look at that after we go through the logic!